rosplan_planning_system/include/rosplan_action_interface/RPActionInterface.h

```diff
@@ -26,10 +26,13 @@ namespace KCL_rosplan {
 	{
 
 	private:
+
+		std::map<std::string, rosplan_knowledge_msgs::DomainFormula> predicates;
+		std::map<std::string, rosplan_knowledge_msgs::DomainFormula> sensed_predicates;
```

Comment:
Any specific reason for this? It may potentially break inherited classes that are currently working now.

Reply:
Yes. The effects list that modifies the knowledge base with the add/remove effects of an action uses the predicates array (same as before). Predicates that are set as sensed are now not set this way.

Note that the members you're highlighting are both private - so it shouldn't affect any inheriting classes.

Reply:
My comment was because before predicates was protected instead of private, and now it's private. So before inherited classes had access to the field, but not anymore.

Reply:
That's true.I can't think of a good reason to keep it private over protected.